src/modules/git_commit.rs

```diff
@@ -0,0 +1,52 @@
+use super::{Context, Module, RootModuleConfig};
+use git2::Repository;
+
+use crate::configs::git_commit::GitCommitConfig;
+
+pub fn id_to_hex_abbrev(bytes: &[u8], len: usize) -> String {
+    bytes
+        .iter()
+        .map(|b| format!("{:02x}", b))
+        .take(len)
+        .collect::<Vec<String>>()
+        .join("")
+}
+
+/// Creates a module with the Git commit in the current directory
+///
+/// Will display the commit hash if the current directory is a git repo
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("git_commit");
+    let config = GitCommitConfig::try_load(module.config);
+    module
+        .get_prefix()
+        .set_value(config.prefix)
+        .set_style(config.style);
+    module
+        .get_suffix()
+        .set_value(config.suffix)
+        .set_style(config.style);
+    module.set_style(config.style);
+
+    let repo = context.get_repo().ok()?;
+    let repo_root = repo.root.as_ref()?;
+    let git_repo = Repository::open(repo_root).ok()?;
+
+    let git_head = git_repo.head();
+    if let Ok(git_head) = git_head {
+        let head_commit = git_head.peel_to_commit().ok()?;
+        let commit_oid = head_commit.id();
+        module.create_segment(
+            "hash",
+            &config
+                .hash
+                .with_value(&id_to_hex_abbrev(commit_oid.as_bytes(), 7)),
+        );
+    }
+
+    if module.is_empty() {
```

Comment:
Can you think of a situation where this code path can be hit?
I'm having trouble thinking of any, if they've made it to here.